test/drop.js

```diff
@@ -18,6 +18,13 @@ describe('drop', function() {
     assert.deepEqual(R.drop(-Infinity, [1, 2, 3]), [1, 2, 3]);
   });
 
+  it('dispatches to `slice` method', function() {
+    var o = {slice: function(a, b) { return '[' + a + ':' + b + ']'; }};
+
+    assert.strictEqual(R.drop(3, 'Ramda'), 'da');
+    assert.strictEqual(R.drop(3, o), '[3:Infinity]');
+  });
+
```

Comment:
to satisfy https://github.com/ramda/ramda/issues/1089 should probably have a test of that sort